modules/hdf4_handler/HDFDMRArray_EOS2LL.cc

```diff
@@ -0,0 +1,283 @@
+    if (gridfd <0) {
+        ostringstream eherr;
+        err_msg = "HDF-EOS: GDopen failed";
+        throw InternalErr (__FILE__, __LINE__, err_msg);
+    }
+
+    int32 gridid = GDattach(gridfd, const_cast<char *>(gridname.c_str()));
+    if (gridid <0) {
+        ostringstream eherr;
+        err_msg = "HDF-EOS: GDattach failed to attach " + gridname;
+        GDclose(gridfd);
+        throw InternalErr (__FILE__, __LINE__, err_msg);
+    }
+
+    // Declare projection code, zone, etc grid parameters. 
+    int32 projcode = -1; 
+    int32 zone     = -1;
+    int32 sphere   = -1;
+    float64 params[16];
+
+    int32 xdim = 0;
+    int32 ydim = 0;
+
+    float64 upleft[2];
+    float64 lowright[2];
+
+    if (GDprojinfo (gridid, &projcode, &zone, &sphere, params) <0) {
+        GDdetach(gridid);
+        GDclose(gridfd);
+        err_msg = "GDprojinfo failed for grid name: " + gridname;
+        throw InternalErr (__FILE__, __LINE__, err_msg);
+    }
+
+    // Retrieve dimensions and X-Y coordinates of corners
+    if (GDgridinfo(gridid, &xdim, &ydim, upleft,
+                   lowright) == -1) {
+        GDdetach(gridid);
+        GDclose(gridfd);
+        err_msg = "GDgridinfo failed for grid name: " + gridname;
+        throw InternalErr (__FILE__, __LINE__, err_msg);
+    }
+ 
+    // Retrieve pixel registration information 
+    int32 pixreg = 0; 
+    intn r = GDpixreginfo (gridid, &pixreg);
+    if (r != 0) {
+        GDdetach(gridid);
+        GDclose(gridfd);
+        err_msg = "GDpixreginfo failed for grid name: " + gridname;
+        throw InternalErr (__FILE__, __LINE__, err_msg);
+    }
+
+    //Retrieve grid pixel origin 
+    int32 origin = 0; 
+    r = GDorigininfo (gridid, &origin);
+    if (r != 0) {
+        GDdetach(gridid);
+        GDclose(gridfd);
+        err_msg = "GDoriginfo failed for grid name: " + gridname;
+        throw InternalErr (__FILE__, __LINE__, err_msg);
+    }
+
+    vector<int32>rows;
+    vector<int32>cols;
+    vector<float64>lon;
+    vector<float64>lat;
+    rows.resize(xdim*ydim);
+    cols.resize(xdim*ydim);
+    lon.resize(xdim*ydim);
+    lat.resize(xdim*ydim);
+
+
+    int i = 0;
+    int j = 0;
+    int k = 0; 
+
+    /* Fill two arguments, rows and columns */
+    // rows             cols
+    //   /- xdim  -/      /- xdim  -/
+    //   0 0 0 ... 0      0 1 2 ... x
+    //   1 1 1 ... 1      0 1 2 ... x
+    //       ...              ...
+    //   y y y ... y      0 1 2 ... x
+
+    for (k = j = 0; j < ydim; ++j) {
+        for (i = 0; i < xdim; ++i) {
+            rows[k] = j;
+            cols[k] = i;
+            ++k;
+        }
+    }
+
+
+    r = GDij2ll (projcode, zone, params, sphere, xdim, ydim, upleft, lowright,
+                 xdim * ydim, rows.data(), cols.data(), lon.data(), lat.data(), pixreg, origin);
+
+    if (r != 0) {
+        GDdetach(gridid);
+        GDclose(gridfd);
+        err_msg = "cannot calculate grid latitude and longitude for grid name: " + gridname;
```

Comment:
I think this err_msg value is lost. Should this be a place where an exception is thrown?

Reply:
You are right. I add the exception.